### ISSUE TITLE
Allow stacking data sets in Line Chart

### ISSUE THREAD

**Issue by KarthikRIyer** [KarthikRIyer/swiftplot#41]
A stack plot is similar to a Line Plot, the difference being that the data sets are stacked over one another. A simple example can be found [here](https://matplotlib.org/gallery/lines_bars_and_markers/stackplot_demo.html#sphx-glr-gallery-lines-bars-and-markers-stackplot-demo-py).

This task will require you to implement the following:
- Add functions to atleast two from the three renderers (AGG/SVG/CoreGraphics) to allow drawing of solid polygons.
- Add the capability to allow stacking of data sets in Line Chart.

**Comment by boronhub** [comment]
Hi, I'm interested in working on this.

**Comment by KarthikRIyer** [comment]
@boronhub that's great! You should claim the task on the GCI website and also state there that you're interested in working on this issue. If you have any other questions, feel free to ask.

**Comment by boronhub** [comment]
One of my other tasks still hasn't been reviewed, so I can't claim it yet. Can I still start work on this so as to familiarize myself with SwiftPlot ?

**Comment by KarthikRIyer** [comment]
@boronhub sure!

**Comment by boronhub** [comment]
Is there a Google Group or an IRC where we can discuss this ?

**Comment by KarthikRIyer** [comment]
There is a Google Group for Swift for TensorFlow, but I prefer we discuss it here in GitHub issues.

**Comment by boronhub** [comment]
The changes have to be made in LineChart.swift, right ?

**Comment by KarthikRIyer** [comment]
@boronhub yes. You need to make changes in LineChart.swift.
You can create a file LineChartSeriesOptions.swift similar to HistogramSeriesOptions.swift. Using that you can decide if you want the plot to be stacked or not.

**Comment by KarthikRIyer** [comment]
@karwa do you have any suggestions regarding this?

**Comment by karwa** [comment]
Not particularly - just to follow the example set by `BarGraph` when it comes to stacking.

Matplotlib calls this a [`stackplot`](https://matplotlib.org/examples/pylab_examples/stackplot_demo.html) and has this as a separate kind of chart. I think that would probably also make sense for us, rather than combining it with the line chart (for example, you can't have multiple series on a line chart if the some of the series also have stacks and are filled underneath).

**Comment by KarthikRIyer** [comment]
I agree. @boronhub you could make a new chart type. I think you should be able to lift most of the code from line chart and make the required changes to get it working.

**Comment by boronhub** [comment]
The example notebook isn't working in Jupyter.
It's saying Line magic command '%install' not found.

**Comment by KarthikRIyer** [comment]
@boronhub the example notebook isn't updated. Are you using swift-jupyter, or colab?

**Comment by boronhub** [comment]
colab

**Comment by KarthikRIyer** [comment]
Plotting on colab won't work at the moment. There's some problem in that. But you can use swift-jupyter. The notebook example hasn't been updated. You can copy the code from the example in the docs.

**Comment by WilliamHYZhang** [comment]
@boronhub are you still working on this? I got swiftplot on Linux working if using the notebook is giving you difficulty.

**Comment by boronhub** [comment]
My VM keeps crashing so im trying to build swift on Windows. I got the notebook working, but %install still isn't working.

**Comment by KarthikRIyer** [comment]
@boronhub you can try using the Linux Subsystem for Windows. So you won't need a VM to use Linux. Check this out: https://docs.microsoft.com/en-us/windows/wsl/install-win10

But you won't be able to use swift-jupyter or any GUI based applications, but you'll be able to work on SwiftPlot and won't need to build swift.

**Comment by boronhub** [comment]
Yeah , I tried WSL as well but to no avail.

**Comment by boronhub** [comment]
I'll figure it out soon

**Comment by WilliamHYZhang** [comment]
Alright, best of luck!

**Comment by boronhub** [comment]
`Installing packages:
	.package(url: "https://github.com/IBM-Swift/BlueCryptor.git", from: "1.0.28")
		Cryptor
	.package(url: "https://github.com/KarthikRIyer/swiftplot", .branch("master"))
		CoreFoundation
		AGGRenderer
		SwiftPlot
With SwiftPM flags: ['-Xcc', '-isystem/usr/include/freetype2', '-Xswiftc', '-lfreetype']
Working in: /tmp/tmp9teg7s8l/swift-install
/tmp/tmp9teg7s8l/swift-install/package: error: manifest parse error(s):
/mnt/e/Dhruv/Projects_19-20/Google_Code_In/Tensorflow/swiftplot/swift-jupyter-master/toolchain/usr/lib/swift/CoreFoundation/CoreFoundation.h:25:10: note: while building module 'SwiftGlibc' imported from /mnt/e/Dhruv/Projects_19-20/Google_Code_In/Tensorflow/swiftplot/swift-jupyter-master/toolchain/usr/lib/swift/CoreFoundation/CoreFoundation.h:25:
#include <sys/types.h>
         ^
<module-includes>:5:10: note: in file included from <module-includes>:5:
#include "/usr/include/utmp.h"
         ^
/usr/include/utmp.h:23:10: note: in file included from /usr/include/utmp.h:23:
#include <sys/types.h>
         ^
//usr/include/x86_64-linux-gnu/sys/types.h:145:10: error: 'stddef.h' file not found
#include <stddef.h>
         ^
<module-includes>:1:10: note: in file included from <module-includes>:1:
#include "CoreFoundation.h"
         ^
/mnt/e/Dhruv/Projects_19-20/Google_Code_In/Tensorflow/swiftplot/swift-jupyter-master/toolchain/usr/lib/swift/CoreFoundation/CoreFoundation.h:25:10: error: could not build module 'SwiftGlibc'
#include <sys/types.h>
         ^
/mnt/e/Dhruv/Projects_19-20/Google_Code_In/Tensorflow/swiftplot/swift-jupyter-master/toolchain/usr/lib/swift/CoreFoundation/CFStream.h:20:10: note: while building module 'CDispatch' imported from /mnt/e/Dhruv/Projects_19-20/Google_Code_In/Tensorflow/swiftplot/swift-jupyter-master/toolchain/usr/lib/swift/CoreFoundation/CFStream.h:20:
#include <dispatch/dispatch.h>
         ^
<module-includes>:1:10: note: in file included from <module-includes>:1:
#include "dispatch.h"
         ^
/mnt/e/Dhruv/Projects_19-20/Google_Code_In/Tensorflow/swiftplot/swift-jupyter-master/toolchain/usr/lib/swift/dispatch/dispatch.h:32:10: note: in file included from /mnt/e/Dhruv/Projects_19-20/Google_Code_In/Tensorflow/swiftplot/swift-jupyter-master/toolchain/usr/lib/swift/dispatch/dispatch.h:32:
#include <os/generic_unix_base.h>
         ^
/mnt/e/Dhruv/Projects_19-20/Google_Code_In/Tensorflow/swiftplot/swift-jupyter-master/toolchain/usr/lib/swift/os/generic_unix_base.h:17:10: note: in file included from /mnt/e/Dhruv/Projects_19-20/Google_Code_In/Tensorflow/swiftplot/swift-jupyter-master/toolchain/usr/lib/swift/os/generic_unix_base.h:17:
#include <sys/sysmacros.h>
         ^
//usr/include/x86_64-linux-gnu/sys/sysmacros.h:39:10: error: could not build module 'SwiftGlibc'
#include <features.h>
         ^
<unknown>:0: error: could not build C module 'CoreFoundation'

Install Error: swift-build returned nonzero exit code 1.

`

**Comment by boronhub** [comment]
error: could not build C module 'CoreFoundation'

can't seem to get over this, any suggestions ?

**Comment by KarthikRIyer** [comment]
@boronhub where are you getting this error? Could you provide more details?

**Comment by boronhub** [comment]
upon executing

%install-swiftpm-flags -Xcc -isystem/usr/include/freetype2 -Xswiftc -lfreetype
%install '.package(url: "https://github.com/IBM-Swift/BlueCryptor.git", from: "1.0.28")' Cryptor
%install '.package(url: "https://github.com/KarthikRIyer/swiftplot", .branch("master"))' SwiftPlot AGGRenderer

in a jupyter notebook (used WSL)

**Comment by KarthikRIyer** [comment]
@boronhub I haven't seen this error before and am having trouble using jupyter notebook in WSL. But you do not need jupyter to work on swiftplot. You can still use the command line.

**Comment by boronhub** [comment]
Is there any documentation for this?

**Comment by WilliamHYZhang** [comment]
Try installing swift on ubuntu linux here: https://itsfoss.com/use-swift-linux (use the most recent swift version files)

**Comment by anigasan** [comment]
@boronhub  - Looks like you are opening in python notebook, %install etc. will not work in Jupyter. I tried installing packages through Anaconda and Python and was not successful.

Alternatively, you can use the route mentioned by @WilliamHYZhang ...It is working! 

Also, I tried using Docker image https://github.com/apple/swift-docker.  It works as well, but requires some minor tweaks.
